src/components/calculator.js

```diff
@@ -1,46 +1,56 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = { total: 0, next: null, operation: null };
+    this.valbutton = this.valbutton.bind(this);
+  }
+
+  valbutton(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
   }
 
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div>
-        <span className="styleoutput">0</span>
+        <span className="styleoutput">
+          {total}
+          {operation}
+          {next}
+        </span>
         <div className="stylebuttoms">
           <div className="styleCalculator">
-            <button type="button">AC</button>
-            <button type="button">7</button>
-            <button type="button">4</button>
-            <button type="button">1</button>
+            <button type="button" onClick={this.valbutton}>AC</button>
+            <button type="button" onClick={this.valbutton}>7</button>
+            <button type="button" onClick={this.valbutton}>4</button>
+            <button type="button" onClick={this.valbutton}>1</button>
           </div>
           <div className="styleCalculator">
-            <button type="button">+/-</button>
-            <button type="button">8</button>
-            <button type="button">5</button>
-            <button type="button">2</button>
+            <button type="button" onClick={this.valbutton}>+/-</button>
+            <button type="button" onClick={this.valbutton}>8</button>
+            <button type="button" onClick={this.valbutton}>5</button>
+            <button type="button" onClick={this.valbutton}>2</button>
           </div>
           <div className="styleCalculator">
-            <button type="button">%</button>
-            <button type="button">9</button>
-            <button type="button">6</button>
-            <button type="button">3</button>
+            <button type="button" onClick={this.valbutton}>%</button>
+            <button type="button" onClick={this.valbutton}>9</button>
+            <button type="button" onClick={this.valbutton}>6</button>
+            <button type="button" onClick={this.valbutton}>3</button>
           </div>
           <div className="styleCalculator">
-            <button className="buttonfunction" type="button">÷</button>
-            <button className="buttonfunction" type="button">*</button>
-            <button className="buttonfunction" type="button">-</button>
-            <button className="buttonfunction" type="button">+</button>
-
+            <button className="buttonfunction" type="button" onClick={this.valbutton}>÷</button>
+            <button className="buttonfunction" type="button" onClick={this.valbutton}>*</button>
```

Comment:
- [ ] Kindly make your app can perform all math operations as required for the project. Right now when clicks, the **multiplication`*`** operator, it blocks the app and shows the error in the image below, but other operations are working as expected, make sure you fix this:
![Screenshot from 2022-10-27 00-30-50](https://user-images.githubusercontent.com/83042742/198151412-2ac1a69a-9b30-499e-8117-2e7bf286455a.png)

Reply:
ok i fix that